app/crud/versions.py

```diff
@@ -72,6 +73,13 @@ async def create_version(dataset: str, version: str, **data) -> ORMVersion:
     if data.get("is_latest"):
         await _reset_is_latest(dataset, version)
 
+    creation_options = data.get("creation_options")
+
+    # if revision, merge with previous
+    if creation_options and creation_options["source_type"] == SourceType.revision:
+        prev_version = await get_version(dataset, creation_options["revision_on"])
+        data["metadata"] = {**data["metadata"], **prev_version.metadata}
```

Comment:
Should this be `data["metadata"] = {**prev_version.metadata, **data["metadata"]}` instead? It looks to me like the old metadata will take precedence over the new, but I could be mis-reading it.